src/ContractionTreeToGraph.jl

```diff
@@ -0,0 +1,120 @@
+"""Function to take a sequence (returned by ITensorNetworks.contraction_sequence) and construct a graph g which represents it (see above)"""
+function create_contraction_tree(contract_sequence)
+    #Collect all the leaves
+    leaves = collect(Leaves(contract_sequence))
+    #Build empty graph
+    g = NamedGraph()
+    #Construct the left branch
+    spawn_child_branch(contract_sequence[1], g, leaves)
+    #Construct the right branch
+    spawn_child_branch(contract_sequence[2], g, leaves)
+
+    #Now we have the vertices we need to figure out the edges
+    for v in vertices(g)
+        #Only add edges from a parent (which defines a tripartition and thus has length 3) to its children
+        if(length(v) == 3)
+        #Work out which vertices it connects to
+            concat1, concat2, concat3 =[v[1]..., v[2]...], [v[2]..., v[3]...], [v[1]..., v[3]...]
+            for vn in setdiff(vertices(g), [v])
+                vn_set = [Set(vni) for vni in vn]
+                if(Set(concat1) ∈ vn_set || Set(concat2) ∈ vn_set || Set(concat3) ∈ vn_set)
+                    add_edge!(g, v => vn)
+                end
+            end
+        end
+    end
+
+
+    return g
+end
+
+"""Given a contraction sequence which is a subsequence of some larger sequence which is being built on current_g and has leaves `leaves`
+Spawn `contract sequence' as a vertex on `current_g' and continue on with its children """
+function spawn_child_branch(contract_sequence, current_g, leaves)
+    #Check if sequence is at leaf point
+    if(isa(contract_sequence, Array))
+        #If not it  will spawn two children
+        group1 = collect(Leaves(contract_sequence[1]))
+        group2 = collect(Leaves(contract_sequence[2]))
+        remaining_verts = setdiff(leaves, vcat(group1, group2))
+        add_vertex!(current_g, (group1, group2, remaining_verts))
+        spawn_child_branch(contract_sequence[1], current_g, leaves)
+        spawn_child_branch(contract_sequence[2], current_g, leaves)
+    else
+        #If it is it is just a vertex
+        add_vertex!(current_g, ([contract_sequence], setdiff(leaves, [contract_sequence])))
+    end
+end
+
+"""Utility functions for the graphical representation of a contraction sequence
+Perhaps it should be a specific object or type of tree?!"""
+
+"""Determine if a node is a leaf"""
+function leaf_node(g::AbstractGraph, v)
+    return length(neighbors(g, v)) == 1
+end
+
+"""Determine if an edge involves a leaf (at src or dst)"""
+function leaf_edge(g::AbstractGraph, e)
+    return leaf_node(g, src(e)) || leaf_node(g, dst(e))
+end 
+
+"""Determine if a node has no neighbours which are leaves"""
+function no_leaf_neighbours(g::AbstractGraph, v)
+    for vn in neighbors(g, v)
+        if(leaf_node(g, vn))
+            return false
+        end
+    end
+    return true
+end
+
+"""Get all edges which do not involve a leaf
+(why does edges(g)[contraction_edge(g, edges(g)) .== true] not work?)"""
+function non_leaf_edges(g::AbstractGraph)
```

Comment:
Maybe `internal_edges`? That would follow the terminology of "internal vertices" and "external vertices": https://en.wikipedia.org/wiki/Tree_(graph_theory)#Tree

Also what about: `filter(e -> !is_leaf_edge(g, e), edges(g))`.